campaign-launcher/interfaceV2/src/api/recordingApiClient.ts

```diff
@@ -0,0 +1,180 @@
+
+type RecordingApiClientConfig = {
+  baseUrl: string;
+  headers?: AxiosRequestHeaders;
+  tokenManager: TokenManager;
+};
+
+export const REFRESH_FAILURE_EVENT = 'refresh-failure';
+
+class HttpError extends Error {
+  public readonly status: number;
+
+  constructor(message: string, status: number) {
+    super(message);
+    this.status = status;
+  }
+
+  static fromAxiosError(error: AxiosError): HttpError {
+    const status = error.response?.status || 0;
+    const message = error.message || 'Request failed';
+    return new HttpError(message, status);
+  }
+} 
+
+export class RecordingApiClient {
+  private axiosInstance: AxiosInstance;
+  private tokenManager: TokenManager;
+  private refreshPromise: RefreshPromise | null = null;
+  
+  constructor(config: RecordingApiClientConfig) {
+    this.axiosInstance = axios.create({
+      baseURL: config.baseUrl,
+      headers: {
+        'Content-Type': 'application/json',
+        ...config.headers,
+      },
+    });
+    this.tokenManager = config.tokenManager;
+
+    this.setupInterceptors();
+  }
+
+  private setupInterceptors(): void {
+    this.axiosInstance.interceptors.request.use(
+      (config: InternalAxiosRequestConfig) => {
+        const accessToken = this.tokenManager.getAccessToken();
+        if (accessToken) {
+          config.headers = config.headers || {};
+          config.headers['Authorization'] = `Bearer ${accessToken}`;
+        }
+        return config;
+      },
+      (error: AxiosError) => Promise.reject(error)
+    );
+
+    this.axiosInstance.interceptors.response.use(
+      (response: AxiosResponse) => response,
+      async (error) => {
+        const originalRequest = error.config;
+        
+        if (error.response?.status === 401) {
+          await this.performRefresh();
+          return this.axiosInstance(originalRequest);
+        }
+
+        throw error;
+      }
+    );
+  }
+
+  private dispatchRefreshFailureEvent(): void {
+    window.dispatchEvent(new Event(REFRESH_FAILURE_EVENT));
+  }
+
+  async performRefresh(): Promise<void> {
+    const refreshToken = this.tokenManager.getRefreshToken();
+    if (!refreshToken) {
+      this.dispatchRefreshFailureEvent();
+      throw new Error('No refresh token');
+    }
+
+    if (!this.refreshPromise){
+      this.refreshPromise = axios.post<TokenData>(`${this.axiosInstance.defaults.baseURL}/auth/refresh`, {
+        refresh_token: refreshToken,
+      });
+    }
+
+    try {
+      const response = await this.refreshPromise;
+      const tokens = {
+        access_token: response.data.access_token,
+        refresh_token: response.data.refresh_token,
+      };
+  
+      this.tokenManager.setTokens(tokens);
+    } catch(e) {
+      if (e instanceof AxiosError && e.response?.status === 401) {
+        this.tokenManager.clearTokens();
+        this.dispatchRefreshFailureEvent();
+        throw new HttpError('Refresh token invalid', 401);
+      }
+      console.error('Refresh token error');
```

Comment:
```suggestion
      console.error('Error while performing tokens refresh', e);
```
It's not `AxiosError` here, so should be safe to log it for debug purpose